xlators/cluster/afr/src/afr-common.c

```diff
@@ -4109,63 +4109,18 @@ afr_release(xlator_t *this, fd_t *fd)
     return 0;
 }
 
-afr_fd_ctx_t *
-__afr_fd_ctx_get(fd_t *fd, xlator_t *this)
-{
-    uint64_t ctx = 0;
-    int ret = 0;
-    afr_fd_ctx_t *fd_ctx = NULL;
-
-    ret = __fd_ctx_get(fd, this, &ctx);
-
-    if (ret < 0) {
-        ret = __afr_fd_ctx_set(this, fd);
-        if (ret < 0)
-            goto out;
-
-        ret = __fd_ctx_get(fd, this, &ctx);
-        if (ret < 0)
-            goto out;
-    }
-
-    fd_ctx = (afr_fd_ctx_t *)(long)ctx;
-out:
-    return fd_ctx;
-}
-
-afr_fd_ctx_t *
-afr_fd_ctx_get(fd_t *fd, xlator_t *this)
-{
-    afr_fd_ctx_t *fd_ctx = NULL;
-
-    LOCK(&fd->lock);
-    {
-        fd_ctx = __afr_fd_ctx_get(fd, this);
-    }
-    UNLOCK(&fd->lock);
-
-    return fd_ctx;
-}
-
-int
+static afr_fd_ctx_t *
 __afr_fd_ctx_set(xlator_t *this, fd_t *fd)
 {
     afr_private_t *priv = NULL;
     int ret = -1;
-    uint64_t ctx = 0;
     afr_fd_ctx_t *fd_ctx = NULL;
     int i = 0;
 
     VALIDATE_OR_GOTO(this->private, out);
-    VALIDATE_OR_GOTO(fd, out);
```

Comment:
@aspandey Why is this check removed?

Reply:
It is  not needed, as this whole call is (now clearly) under fd->lock, so fd cannot be NULL. Please refer the previous comment and discussion also.

Reply:
Ah, missed it. Thanks for pointing it out.